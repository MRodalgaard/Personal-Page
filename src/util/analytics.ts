import 'firebase/analytics';
import firebase from './firebase';

let analytics: firebase.analytics.Analytics | undefined;

export enum LogEvent {
  ColorClick = 'color_click',
  QuoteClick = 'quote_click',
}

class Analytics {
  constructor() {
    firebase.analytics.isSupported().then((isSupported) => {
      if (isSupported) {
        analytics = firebase.analytics();
      }
    });
  }

  public logEvent = (event: LogEvent, parameters?: any) => {
    analytics?.logEvent(event, parameters);
  };
}

export default new Analytics();
