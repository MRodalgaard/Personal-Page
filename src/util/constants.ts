// Key for persisting app state in local storage
export const LOCALSTORAGE_KEY = "AppState";

export enum AppBackground {
  default,
  greyscale,
  colored
}
