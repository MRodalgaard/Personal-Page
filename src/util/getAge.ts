const getAge = (birthday: string): number | undefined => {
  const today = new Date();
  const birthDate = new Date(birthday);

  if (isNaN(birthDate.getTime()) || today < birthDate) {
    return undefined;
  }

  let age = today.getFullYear() - birthDate.getFullYear();
  const m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
    age--;
  }
  return age;
};

export default getAge;
