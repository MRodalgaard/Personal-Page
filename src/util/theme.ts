export enum AppColors {
  primary = '#000000',
  secondary = '#c9282d',
  grey = '#333333',
}
