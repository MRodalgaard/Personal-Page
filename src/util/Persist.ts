import { IAppState } from '../components/App/AppContext';
import { LOCALSTORAGE_KEY } from './constants';

const Persist = {
  persistState: (state: IAppState) => {
    localStorage.setItem(LOCALSTORAGE_KEY, JSON.stringify(state));
  },

  getPersistedState: (): IAppState | undefined => {
    const jsonState = localStorage.getItem(LOCALSTORAGE_KEY);
    return jsonState ? JSON.parse(jsonState) : undefined;
  },
};

export default Persist;
