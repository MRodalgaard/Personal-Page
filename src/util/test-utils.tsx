import { render, RenderOptions } from '@testing-library/react';
import AppProvider from 'components/App/AppProvider';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

type Props = {
  children: ReactNode;
};

const AllTheProviders = ({ children }: Props) => {
  return (
    <AppProvider>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </AppProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, {
    wrapper: AllTheProviders as React.FunctionComponent,
    ...options,
  });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
