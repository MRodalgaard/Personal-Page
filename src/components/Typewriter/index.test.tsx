import React from 'react';
import { render } from 'util/test-utils';
import { default as Typewriter } from '.';

describe('Typewriter', () => {
  it('renders', () => {
    const { container } = render(
      <Typewriter>
        <p>Hello</p>
        <p>World</p>
      </Typewriter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
