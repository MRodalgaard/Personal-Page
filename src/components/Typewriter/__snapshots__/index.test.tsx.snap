// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typewriter renders 1`] = `
.c2 {
  color: #000000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  -webkit-transition: color 0.5s ease;
  transition: color 0.5s ease;
}

.c2:hover {
  color: #c9282d;
}

.c1 {
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  font-size: 12px;
  padding-left: 6px;
  -webkit-animation: blink 0.5s linear infinite alternate;
  animation: blink 0.5s linear infinite alternate;
}

.c4 {
  opacity: 0.06;
}

.c0 {
  display: contents;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      aria-label="volume"
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-volume-mute fa-w-16 "
        data-icon="volume-mute"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M215.03 71.05L126.06 160H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V88.02c0-21.46-25.96-31.98-40.97-16.97zM461.64 256l45.64-45.64c6.3-6.3 6.3-16.52 0-22.82l-22.82-22.82c-6.3-6.3-16.52-6.3-22.82 0L416 210.36l-45.64-45.64c-6.3-6.3-16.52-6.3-22.82 0l-22.82 22.82c-6.3 6.3-6.3 16.52 0 22.82L370.36 256l-45.63 45.63c-6.3 6.3-6.3 16.52 0 22.82l22.82 22.82c6.3 6.3 16.52 6.3 22.82 0L416 301.64l45.64 45.64c6.3 6.3 16.52 6.3 22.82 0l22.82-22.82c6.3-6.3 6.3-16.52 0-22.82L461.64 256z"
          fill="currentColor"
        />
      </svg>
    </a>
    <a
      aria-label="play"
      class="c2 c3"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
  <p>
    
    <span
      class="c4"
      opacity="0.06"
    >
      Hello
    </span>
  </p>
  <p>
    <span
      class="c4"
      opacity="0.06"
    >
      World
    </span>
  </p>
</div>
`;
